.github/eval.sh

```diff
@@ -0,0 +1,182 @@
+                fi
+                ;;
+            *)
+                print_err "unrecognized option"; usage; exit 1 ;;
+        esac
+    done
+    shift $((OPTIND-1))
+    if [ -n "$*" ]; then
+        print_err "unsupported positional argument(s): '$*'"; exit 1
+    fi
+    if [ -z "$JOB_ID" ]; then
+        print_err "missing mandatory option (-j)"; usage; exit 1
+    fi
+    if [ -z "$MAX_JOBS" ]; then
+        print_err "missing mandatory option (-m)"; usage; exit 1
+    fi
+    if [ "$JOB_ID" -ge "$MAX_JOBS" ]; then
+        print_err "'-j' must be smaller than '-m'"; usage; exit 1
+    fi
+}
+
+exit_unless_command_exists () {
+    if ! command -v "$1" &>/dev/null; then
+        print_err "command '$1' is not installed (Hint: are you inside a nix-shell?)"
+        exit 1
+    fi
+}
+
+################################################################################
+
+evaluate () {
+    job="$1"
+    max_jobs="$2"
+    filter="$3"
+    echo "[+] Using filter: '$filter'"
+    # Output all flake output names
+    nix flake show --all-systems --json |\
+      jq  '[paths(scalars) as $path | { ($path|join(".")): getpath($path) }] | add' \
+      >"$TMPDIR/outs_all.txt"
+    # Apply the given filter
+    if ! grep -Po "${filter}\S*.name" "$TMPDIR/outs_all.txt" >"$TMPDIR/outs.txt"; then
+        print_err "No flake outputs match filter: '$filter'"; exit 1
+    fi
+    # Remove the '.name' suffix
+    sed -i "s/.name//" "$TMPDIR/outs.txt"
+    # Read the attribute set names
+    grep -oP ".*(?=\.)" "$TMPDIR/outs.txt" | sort | uniq >"$TMPDIR/attrs.txt"
+    # Generate eval expression on the fly
+    printf '%s\n' \
+      "let" \
+      "  flake = builtins.getFlake ("git+file://" + toString ./.);"\
+      "  lib = (import flake.inputs.nixpkgs { }).lib;"\
+      "in {" >"$TMPDIR/eval.nix"
+    while read -r attrset; do
+        mapfile -t attrs < <(grep "$attrset" "$TMPDIR/outs.txt" | rev | cut -d '.' -f1 | rev | sort | uniq)
+        # Split the target attribute set so the evaluation work gets distributed
+        # somewhat evenly between 'max_jobs' runners
+        split_size=$(( ( ${#attrs[@]} + max_jobs - 1 ) / max_jobs ))
+        # Select the attributes that will be evaluated by this runner
+        start_index=$(( job * split_size ))
+        split_attrs=("${attrs[@]:$start_index:$split_size}")
+        if [ "${#split_attrs[@]}" -eq 0 ]; then
+            continue
+        fi
+        # shellcheck disable=SC2129
+        printf "  out_%s = lib.getAttrs [ " "$attrset" >>"$TMPDIR/eval.nix"
+        printf " \"%s\" "  "${split_attrs[@]}" >>"$TMPDIR/eval.nix"
+        printf " ] flake.%s;\n" "$attrset" >>"$TMPDIR/eval.nix"
+    done < "$TMPDIR/attrs.txt"
+    printf "}\n" >>"$TMPDIR/eval.nix"
+    echo "[+] Evaluating nix expression:"
+    cat "$TMPDIR/eval.nix"
+    # Evaluate with nix-eval-jobs
+    gcroot="$TMPDIR/gcroot"
+    nix-eval-jobs \
+      --accept-flake-config \
+      --gc-roots-dir "$gcroot" \
+      --force-recurse \
```

Comment:
```suggestion
      --force-recurse \
      --check-cache-status \
```

Reply:
The actual build has not been implemented yet?

Reply:
Do you mean pre-merge ghaf builds?
They have been runnnig for quite some time, see: https://github.com/tiiuae/ghaf/actions/workflows/build.yml